parquet-column/src/main/java/org/apache/parquet/column/values/bloomfilter/BlockSplitBloomFilter.java

```diff
@@ -74,51 +71,84 @@
   // of bits to set, one per 32-bit word.
-  private static final int SALT[] = {0x47b6137b, 0x44974d91, 0x8824ad5b, 0xa2b7289d,
+  private static final int[] SALT = {0x47b6137b, 0x44974d91, 0x8824ad5b, 0xa2b7289d,
     0x705495c7, 0x2df1424b, 0x9efc4947, 0x5c6bfb31};
 
   /**
-   * Constructor of Bloom filter.
+   * Constructor of block-based Bloom filter.
    *
    * @param numBytes The number of bytes for Bloom filter bitset. The range of num_bytes should be within
-   *                 [MINIMUM_BYTES, MAXIMUM_BYTES], it will be rounded up/down
+   *                 [DEFAULT_MINIMUM_BYTES, DEFAULT_MAXIMUM_BYTES], it will be rounded up/down
    *                 to lower/upper bound if num_bytes is out of range. It will also be rounded up to a power
-   *                 of 2. It uses murmur3_x64_128 as its default hash function.
+   *                 of 2. It uses XXH64 as its default hash function.
    */
   public BlockSplitBloomFilter(int numBytes) {
-    this(numBytes, HashStrategy.MURMUR3_X64_128);
+    this(numBytes, DEFAULT_MAXIMUM_BYTES, HashStrategy.XXH64);
   }
 
   /**
-   * Constructor of block-based Bloom filter. It uses murmur3_x64_128 as its default hash
-   * function.
+   * Constructor of block-based Bloom filter.
+   *
+   * @param numBytes The number of bytes for Bloom filter bitset. The range of num_bytes should be within
+   *                 [DEFAULT_MINIMUM_BYTES, DEFAULT_MAXIMUM_BYTES], it will be rounded up/down
+   *                 to lower/upper bound if num_bytes is out of range. It will also be rounded up to a power
+   *                 of 2. It uses XXH64 as its default hash function.
+   * @param maximumBytes The maximum bytes of the Bloom filter.
+   */
+  public BlockSplitBloomFilter(int numBytes, int maximumBytes) {
+    this(numBytes, maximumBytes, HashStrategy.XXH64);
+  }
+
+  /**
+   * Constructor of block-based Bloom filter.
    *
    * @param numBytes The number of bytes for Bloom filter bitset
    * @param hashStrategy The hash strategy of Bloom filter.
    */
   private BlockSplitBloomFilter(int numBytes, HashStrategy hashStrategy) {
+    this(numBytes, DEFAULT_MAXIMUM_BYTES, hashStrategy);
+  }
+
+  /**
+   * Constructor of block-based Bloom filter.
+   *
+   * @param numBytes The number of bytes for Bloom filter bitset. The range of num_bytes should be within
+   *                 [DEFAULT_MINIMUM_BYTES, maximumBytes], it will be rounded up/down to lower/upper bound if
+   *                 num_bytes is out of range. It will also be rounded up to a power of 2.
+   * @param maximumBytes The maximum bytes of the Bloom filter.
+   * @param hashStrategy The adopted hash strategy of the Bloom filter.
+   */
+  public BlockSplitBloomFilter(int numBytes, int maximumBytes, HashStrategy hashStrategy) {
+    if (maximumBytes > DEFAULT_MINIMUM_BYTES) {
+      this.maximumBytes = maximumBytes;
+    }
     initBitset(numBytes);
+
     switch (hashStrategy) {
-      case MURMUR3_X64_128:
+      case XXH64:
```

Comment:
Can we add a comment here why we chose this algorithm? And the characteristics of the algorithm.

Reply:
sure